src/pages/ProductPage.js

```diff
@@ -0,0 +1,80 @@
+import { useState, useEffect } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { GetData } from "../API";
+import ProductCard from "../components/ProductCard";
+import ProductFilter, { useCustomFilter } from "../components/ProductFilter";
+import { useUserContext } from "../context & data/UserContext";
+
+const ProductPage = () => {
+  // to get the params from the url
+  const { filterName, catergoryName } = useParams();
+  const navigate = useNavigate();
+
+  //basic products that we got
+  const [productsData, setProductsData] = useState([]);
+  const { state } = useUserContext();
+
+  // this is the setup to use filter (bear with me , this one is nice as AF)
+  // the ProductFilter componenet gets the setfilter as a prop as it handles the filter data, whenever the state of filter in
+  // the ProductFilter componenet changes it run this function and pass down the latest filter configuration.
+  //later this below filter state calls the custom hook for filtering with the present products set and returns a filtered
+  //array of products according to the filter configuration
+  const [filter, setFilter] = useState({});
```

Comment:
use null as initial state if the state is going to be a object instead of empty object i.e `{}`

Reply:
let me check that first, if that would work properly